src/sentry/web/frontend/oauth_token.py

```diff
@@ -46,81 +48,90 @@ def post(self, request: HttpRequest) -> HttpResponse:
         client_id = request.POST.get("client_id")
         if not client_id:
             return self.error(request, "invalid_client", "missing client_id")
-        if grant_type == GrantTypes.AUTHORIZATION:
-            return self._get_access_tokens(request, client_id)
-        elif grant_type == GrantTypes.REFRESH:
-            return self._get_refresh_token(request, client_id)
-        else:
+        if grant_type not in [GrantTypes.AUTHORIZATION, GrantTypes.REFRESH]:
             return self.error(request, "unsupported_grant_type")
-
-    def _get_access_tokens(self, request, client_id):
-        redirect_uri = request.POST.get("redirect_uri")
-        code = request.POST.get("code")
-
         try:
             application = ApiApplication.objects.get(
                 client_id=client_id, status=ApiApplicationStatus.active
             )
         except ApiApplication.DoesNotExist:
             return self.error(request, "invalid_client", "invalid client_id")
 
+        client_secret = request.POST.get("client_secret")
```

Comment:
nit:  can we move this up with the rest of the attr gets?

Reply:
I prefer to define variables as close to where they are used as possible ([reasoning](https://codereview.stackexchange.com/questions/6283/variable-declaration-closer-to-usage-vs-declaring-at-the-top-of-method))

if you feel strongly the other way i can move it up

Reply:
in this case i do feel a bit more strongly

The stack overflow is valid - except we're still in the same scope here (looks like their main reasoning is to keep the declarations pinned to the scopes they're being utilized)
So - I would absolutely agree, if we're passing these args through to other callbacks/scopes - then we should define the args in those scopes individually (hence i think i had a comment somewhere calling out we should deconstruct the request args within the scopes theyre being used and not passing them into the helpers - mightve been in the other refactor PR)